src/Microsoft.ML.EntryPoints/TrainTestSplit.cs

```diff
@@ -106,11 +108,15 @@ public static string CreateStratificationColumn(IHost host, ref IDataView data,
                 else
                 {
                     if (data.Schema[stratificationColumn].IsNormalized() || (type != NumberDataViewType.Single && type != NumberDataViewType.Double))
-                        return stratificationColumn;
-
-                    data = new NormalizingEstimator(host,
-                        new NormalizingEstimator.MinMaxColumnOptions(stratCol, stratificationColumn, ensureZeroUntouched: true))
-                        .Fit(data).Transform(data);
+                    {
+                        data = new ColumnCopyingEstimator(host,(stratCol,stratificationColumn)).Fit(data).Transform(data);
```

Comment:
>ColumnCopyingEstimator [](start = 35, length = 22)

Should this be added to `EnsureGroupPreservationColumn` as well?

And more generally, would it make sense to unify the two methods? There is also a method called `GetSplitColumn` in line 285 in CrossValidationCommand.cs that does more or less the same thing. I think the same utility method should be called for maml, entry points and the C# API.

Reply:
So I took a quick look into the `EnsureGroupPreservationColumn` callers, and it doesn't seem that they drop the created column, so this wouldn't be an issue there. I'll look into it more later to confirm this.

About merging these methods, I think it would be a good idea to maintain consistency across the codebase, although it wouldn't be necessary to fix the issue opened by @ganik to unblock NimbusML update. I'll look into this idea.

---
In reply to: [438552643](https://github.com/dotnet/machinelearning/pull/5227#discussion_r438552643) [](ancestors = 438552643)

Reply:
I confirmed that when `EnsureGroupPreservationColumn` was called, the `samplingKeyColumn` was never dropped, so the issue didn't apply on that case.

Also, I've merged the methods you've mentioned, so now only `CreateGroupPreservationColumn` exists

---
In reply to: [438969558](https://github.com/dotnet/machinelearning/pull/5227#discussion_r438969558) [](ancestors = 438969558,438552643)

Reply:
Update: I've renamed `CreateGroupPreservationColumn` to `CreateSplitColumn` and made similar renames elsewhere to make the code easier to read. Now "splitColumn" is the name of the temporary column we create for splitting, regardless if it's based-off a "samplingKeyColumn" (ML.NET) or a "stratificationColumn" (NimbusML, Maml, legacy TLC naming conventions...)

Also now I do drop the new "splitColumn" when it's created while splitting through ML.NET's API, because not dropping it was causing other issues... see comment:
[https://github.com/dotnet/machinelearning/pull/5227#discussion_r446485476](https://github.com/dotnet/machinelearning/pull/5227#discussion_r446485476)

---
In reply to: [441992076](https://github.com/dotnet/machinelearning/pull/5227#discussion_r441992076) [](ancestors = 441992076,438969558,438552643)